pkg/api/login_oauth.go

```diff
@@ -76,28 +76,30 @@ func (hs *HTTPServer) OAuthLogin(ctx *models.ReqContext) response.Response {
 	loginInfo.AuthModule = name
 	provider := hs.SocialService.GetOAuthInfoProvider(name)
 	if provider == nil {
+		pMsg := "OAuth not enabled"
 		hs.handleOAuthLoginError(ctx, loginInfo, LoginError{
 			HttpStatus:    http.StatusNotFound,
-			PublicMessage: "OAuth not enabled",
+			PublicMessage: pMsg,
 		})
-		return nil
+		return response.Error(http.StatusNotFound, pMsg, errors.New(pMsg))
 	}
 
 	connect, err := hs.SocialService.GetConnector(name)
 	if err != nil {
+		pMsg := fmt.Sprintf("No OAuth with name %s configured", name)
 		hs.handleOAuthLoginError(ctx, loginInfo, LoginError{
 			HttpStatus:    http.StatusNotFound,
-			PublicMessage: fmt.Sprintf("No OAuth with name %s configured", name),
+			PublicMessage: pMsg,
 		})
-		return nil
+		return response.Error(http.StatusNotFound, pMsg, err)
 	}
 
 	errorParam := ctx.Query("error")
 	if errorParam != "" {
 		errorDesc := ctx.Query("error_description")
 		oauthLogger.Error("failed to login ", "error", errorParam, "errorDesc", errorDesc)
 		hs.handleOAuthLoginErrorWithRedirect(ctx, loginInfo, login.ErrProviderDeniedRequest, "error", errorParam, "errorDesc", errorDesc)
-		return nil
+		return response.Error(http.StatusInternalServerError, login.ErrProviderDeniedRequest.Error(), err)
```

Comment:
I think everytime there was a call to `hs.handleOAuthLoginErrorWithRedirect` the response is meant to be `response.Redirect`

Reply:
I suppose we can redirect to /login 👍 but halting any change for now, until we decide on the right approach